test/miner_txn_mgr_SUITE.erl

```diff
@@ -358,7 +366,433 @@ txn_dependent_test(Config) ->
+    %% which are shared across all 3 queue
+    %% tests ( one for each of the txn protocols )
+    Config1 = txn_queue_common_test(Config),
+    %% the common test appends a shared dataset
+    %% to the config
+    %% take what we need from it to complete
+    %% the rest of this test
+    CommonDataMap = ?config(common_data, Config1),
+    #{
+        txn1_key := Txn1Key,
+        txn2_key := Txn2Key,
+        txn3_key := Txn3Key,
+        txn4_key := Txn4Key,
+        txn1_submit_height := Txn1SubmitHeight,
+        txn2_submit_height := Txn2SubmitHeight,
+        txn3_submit_height := Txn3SubmitHeight,
+        txn4_submit_height := Txn4SubmitHeight,
+        con_miner_1 := ConMiner1,
+        miner := Miner,
+        last_height :=  Height
+    } = CommonDataMap,
+    ConMiner1Addr = miner_ct_utils:node2addr(ConMiner1, AddrList),
+
+    %% query the txn mgr txn status API
+    %% queue data will not be available and will default to zero
+    ok = miner_ct_utils:wait_for_gte(height_exactly, Miners, Height + 3),
+    {ok, Txn1Key, _, Txn1SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn1Key, Miner),
+    {ok, Txn2Key, _, Txn2SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn2Key, Miner),
+    {ok, Txn3Key, _, Txn3SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn3Key, Miner),
+    {ok, Txn4Key, _, Txn4SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn4Key, Miner),
+
+    ok.
+
+txn_queue_common_test(Config) ->
+    %% this sets up and runs the elements of the
+    %% txn queue tests which are common
+    %% across all 3 protocol versions
+    %% any data required by the parent
+    %% will be added to the config
+
+    Miners = ?config(miners, Config),
+    NonConMiners = ?config(non_consensus_miners, Config),
+    Miner = hd(NonConMiners),
+    ConMiners = ?config(consensus_miners, Config),
+    ConMiner1 = hd(ConMiners),
+    AddrList = ?config(tagged_miner_addresses, Config),
+
+    Addr = miner_ct_utils:node2addr(Miner, AddrList),
+    ConMiner1Addr = miner_ct_utils:node2addr(ConMiner1, AddrList),
+
+    Chain = ct_rpc:call(Miner, blockchain_worker, blockchain, []),
+    ct:pal("consensus miners ~p", [ConMiners]),
+    ct:pal("non consensus miners ~p", [NonConMiners]),
+    ct:pal("miner in use ~p", [Miner]),
+    ct:pal("consensus miner in use ~p", [ConMiner1]),
+    ct:pal("consensus miner bin in use ~p", [ConMiner1Addr]),
+
+    PayerAddr = Addr,
+    Payee = hd(miner_ct_utils:shuffle(ConMiners)),
+    PayeeAddr = miner_ct_utils:node2addr(Payee, AddrList),
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Miner, blockchain_swarm, keys, []),
+    IgnoredTxns = [],
+
+    %% meck out the sidecar random_n function
+    %% make it return an empty list
+    %% simulate an empty buffer
+    %% this should result in the submitted txns
+    %% remaining in the sidecar buffer
+    meck:new(hbbft_utils, [passthrough]),
```

Comment:
i'm surprised you can tell the remote rpc nodes to expect a meck intercept of a function call when you've only created the meck process on the test running node/process

Reply:
i too was surprised but thankful